api/src/database/index.ts

```diff
@@ -243,3 +247,21 @@ export async function validateDatabaseExtensions(): Promise<void> {
 		}
 	}
 }
+
+async function validateDatabaseCharset(database?: Knex): Promise<void> {
+	database = database ?? getDatabase();
+	if (getDatabaseClient(database) === 'mysql') {
+		const directusResult = await database('information_schema.columns')
+			.select(`collation_name as collation`)
+			.where({ table_name: 'directus_users', column_name: 'id' })
+			.first();
```

Comment:
Lets check against all the Directus system columns, instead of just `directus_users.id`